README.md

```diff
@@ -6,6 +6,62 @@ This repository contains documentation for the MongoDB Kotlin Driver.
 
 Please file issue reports or requests at the [Documentation Jira Project](https://jira.mongodb.org/browse/DOCS).
 
+## Building API Documentation
+
+Build the API documentation docs for any major and minor version releases.
+
+We use [Dokka](https://github.com/Kotlin/dokka) to generate the API documentation and host them on GitHub Pages.
+
+To execute a build, you must have Java JDK 17+ installed. You can use brew to install jEnv and OpenJDK to manage your Java versions:
+
+```sh
+brew install jenv
+brew install openjdk@17
+```
+
+The output of the `openjdk@17` includes a command that starts with `sudo ln -sfn`. Execute this command and then run the following commands to add the JDK to your environment:
+
+```sh
+jenv add /Library/Java/JavaVirtualMachines/openjdk-17.jdk/Contents/Home/
+jenv global 17.0
+```
+
+Fork and clone the [`mongo-java-driver`](https://github.com/mongodb/mongo-java-driver) repository. Navigate to the repository and run the following `gradle` command:
+
+```sh
+./gradlew dokkaHtml
+```
+
+This should generate the docs in `./build/docs/`
+Check out the upstream `gh-pages` branch:
+
+```sh
+git checkout gh-pages
+```
+
+If the branch doesn't exist, create a directory for the target version. For example:
+
+```sh
+mkdir -p 4.10/apidocs/
+```
+
+Move the contents of the `./build/docs/` directory into the target version directory. For example:
+
+```sh
+mv ./build/docs/* 4.10/apidocs/
+```
+
+Stage the newly-created docs in Git and ensure that you did not stage anything but the desired pages. For example,
+
+```sh
+git add 4.10/apidocs
+git status
+```
+
+Commit and push your changes for review. Once the PR is merged to the upstream `gh-pages` branch, GitHub Pages will automatically deploy the API documentation.
```

Comment:
Suggestion:

I think this should be something like "Commit and push your changes to your feature branch. Then, create a PR that compares changes against the original repository's gh-pages branch. After you get approval and merge the PR, your changes are automatically published to GitHub Pages."